src/main/java/io/tiledb/spark/TileDBDataReaderPartitionScan.java

```diff
@@ -530,11 +551,19 @@ private int getColumnBatch(StructField field, int index) throws TileDBError {
     }
   }
 
-  private int getScalarValueColumn(String name, Datatype dataType, int index) throws TileDBError {
-
-    metricsUpdater.startTimer(queryGetScalarAttributeTimerName);
-    int numValues;
+  /**
+   * Puts the non-null values in the resultVector.
+   *
+   * @param name name of the attribute
+   * @param dataType datatype of the attribute
+   * @param index index of the resultVectorArray
+   * @return
+   * @throws TileDBError
+   */
+  private int putNonNullValuesOfNullableAttr(String name, Datatype dataType, int index)
```

Comment:
We should probably consider changing the name of this method, as it is used both for nullable and non-nullable attributes.

Reply:
Done